bitcoin/examples/ecdsa-psbt-simple.rs

```diff
@@ -0,0 +1,246 @@
+        vout: 1,
+    };
+
+    let utxo_2 = TxOut { value: DUMMY_UTXO_AMOUNT_INPUT_2, script_pubkey: script_pubkey_2 };
+    vec![(out_point_1, utxo_1), (out_point_2, utxo_2)]
+}
+
+fn main() {
+    let secp = Secp256k1::new();
+
+    // Get the individual xprivs we control. In a real application these would come from a stored secret.
+    let master_xpriv = Xpriv::from_str(XPRIV).expect("valid xpriv");
+    let xpriv_input_1 = get_external_address_xpriv(&secp, master_xpriv, 0);
+    let xpriv_input_2 = get_internal_address_xpriv(&secp, master_xpriv, 0);
+    let xpriv_change = get_internal_address_xpriv(&secp, master_xpriv, 1);
+
+    // Get the PKs
+    let pk_input_1 = Xpub::from_priv(&secp, &xpriv_input_1).to_pub();
+    let pk_input_2 = Xpub::from_priv(&secp, &xpriv_input_2).to_pub();
+    let pk_inputs = [pk_input_1, pk_input_2];
+    let pk_change = Xpub::from_priv(&secp, &xpriv_change).to_pub();
+
+    // Get the Witness Public Key Hashes (WPKHs)
+    let wpkhs: Vec<WPubkeyHash> = pk_inputs.iter().map(|pk| pk.wpubkey_hash()).collect();
+
+    // Get the unspent outputs that is locked to the key above that we control.
+    // In a real application these would come from the chain.
+    let utxos: Vec<TxOut> =
+        dummy_unspent_transaction_outputs().into_iter().map(|(_, utxo)| utxo).collect();
+
+    // Get the addresses to send to.
+    let address = receivers_address();
+
+    // The inputs for the transaction we are constructing.
+    let inputs: Vec<TxIn> = dummy_unspent_transaction_outputs()
+        .into_iter()
+        .map(|(outpoint, _)| TxIn {
+            previous_output: outpoint,
+            script_sig: ScriptBuf::default(),
+            sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+            witness: Witness::default(),
+        })
+        .collect();
+
+    // The spend output is locked to a key controlled by the receiver.
+    let spend = TxOut { value: SPEND_AMOUNT, script_pubkey: address.script_pubkey() };
+
+    // The change output is locked to a key controlled by us.
+    let change = TxOut {
+        value: CHANGE_AMOUNT,
+        script_pubkey: ScriptBuf::new_p2wpkh(&pk_change.wpubkey_hash()), // Change comes back to us.
+    };
+
+    // The transaction we want to sign and broadcast.
+    let unsigned_tx = Transaction {
+        version: transaction::Version::TWO,  // Post BIP 68.
+        lock_time: absolute::LockTime::ZERO, // Ignore the locktime.
+        input: inputs,                       // Input is 0-indexed.
+        output: vec![spend, change],         // Outputs, order does not matter.
+    };
+
+    // Now we'll start the PSBT workflow.
+    // Step 1: Creator role; that creates,
+    // and add inputs and outputs to the PSBT.
+    let mut psbt = Psbt::from_unsigned_tx(unsigned_tx).expect("Could not create PSBT");
+
+    // Step 2:Updater role; that adds additional
+    // information to the PSBT.
+    let ty = PsbtSighashType::from_str("SIGHASH_ALL").expect("valid sighash type");
```

Comment:
```suggestion
    let ty = EcdsaSighashType::All.into();
```

Reply:
And that explains why we don't have consts on the `PsbtSighashType`, took me until now to remember. We should document that.